From 73ae708c19893a1c2e76cda9c1f7caac8e278258 Mon Sep 17 00:00:00 2001
From: Josh Triplett <josh@joshtriplett.org>
Date: Mon, 23 Dec 2013 13:56:06 -0800
Subject: [PATCH 7/7] Makefile: Build with -Werror=date-time if the compiler
 supports it

Patch-mainline: 3.14-rc1
Git-commit: fe7c36c7bde12190341722af69358e42171162f3

GCC 4.9 and newer have a new warning -Wdate-time, which warns on any use
of __DATE__, __TIME__, or __TIMESTAMP__, which would make the build
non-deterministic.  Now that the kernel does not use any of those
macros, turn on -Werror=date-time if available, to keep it that way.

The kernel already (optionally) records this information at build time
in a single place; other kernel code should not duplicate that.

Signed-off-by: Josh Triplett <josh@joshtriplett.org>
Signed-off-by: Michal Marek <mmarek@suse.cz>

Conflicts:
	Makefile

Signed-off-by: Jan Moskyto Matejka <mq@suse.cz>
---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index e41f949..ba32a3b 100644
--- a/Makefile
+++ b/Makefile
@@ -685,6 +685,9 @@ KBUILD_CFLAGS	+= $(call cc-option,-fno-strict-overflow)
 # conserve stack if available
 KBUILD_CFLAGS   += $(call cc-option,-fconserve-stack)
 
+# Prohibit date/time macros, which would make the build non-deterministic
+KBUILD_CFLAGS   += $(call cc-option,-Werror=date-time)
+
 # use the deterministic mode of AR if available
 KBUILD_ARFLAGS := $(call ar-option,D)
 
-- 
1.8.4.5

